### ISSUE TITLE
AutoPlay is paused when re-rendering the Slider Component 

### ISSUE THREAD

**Issue by bahaagalal** [akiran/react-slick#634]
I think there is a bug over here [https://github.com/akiran/react-slick/blob/master/src/mixins/helpers.js#L71](https://github.com/akiran/react-slick/blob/master/src/mixins/helpers.js#L71)

It should be the other way around. 

```javascript
if(props.autoplay) {
      this.autoPlay();
} else {
      this.pause();
}
```

**Comment by LoicMahieu** [comment]
I confirm a problem has been introduced in latest version with autoplay.

**Comment by chaucerbao** [comment]
Bug...or [feature](https://github.com/akiran/react-slick/commit/f68bb3e68a6997aa018fa60c483be06100ec2a98#diff-3f43ed4258c814c46d337afc1a320ae2)?

**Comment by Sylvenas** [comment]
I'm in the same boat,in the process of using react-router,when the route come back,the animation is stop.

**Comment by bahaagalal** [comment]
@chaucerbao I am not sure if this should be considered bug or feature; but I think the expected behavior of the autoplay property is to make the slider auto switch slides unless some explicit event has happened like calling `pause` function or enabling `pauseOnHover`  property. Following the same way of thinking we could add a new property like `pauseOnRerender` that pause the slider if an update operation happened. What do you think?

**Comment by chaucerbao** [comment]
@bahaagalal Oh, I definitely don't disagree with you. I'm here because I'm having the same problem. I was just checking out the commit history and found that a "fix" was *seemingly* implemented for #470, perhaps too quickly.

**Comment by kcd71461** [comment]
i solved this problem by using slickNext method and setTimeout. T.T


    startSlideTimeout() {
        this.timeout = setTimeout(() => {
            this.slider.slickNext();
            this.startSlideTimeout();
        }, 1000);
    }

    endSlideTimeout() {
        clearTimeout(this.timeout);
    }

**Comment by benji06000** [comment]
Thanks for Satazor for the PR : #651

**Comment by FuckingChestnut** [comment]
May be you can use this to resolve the problem:
`setTimeout(slider.innerSlider.autoPlay, 50)`

**Comment by devendra-gh** [comment]
Try this

if(!props.autoplay) {
      this.autoPlay();
} else {
      this.pause();
}

**Comment by jitenderchand1** [comment]
By when we can expect this to be merge in repo

**Comment by mjaltamirano** [comment]
@jitenderchand1 it looks like the CI build failed #713

**Comment by siriwatknp** [comment]
@devendra-gh your workaround still not work because in `this.autoPlay()`

```
autoPlay: function autoPlay() {
    if (this.state.autoPlayTimer) {
      console.log('this.state.autoPlayTimer',this.state.autoPlayTimer)
      clearTimeout(this.state.autoPlayTimer);
    }
    if (this.props.autoplay) {
      this.setState({
        autoPlayTimer: setTimeout(this.play, this.props.autoplaySpeed)
      });
    }
  },
```

because `this.props` is still false `this.setState(...)` will not be called, to make it right I think we should remove `if(this.props.autoplay)` or use `nextProps` instead of `this.props`

**Comment by eHallberg** [comment]
Hey,
I had this issue aswell, when changing size of window or pressing a slide (opening modal) it stopped autoplay.
I have found a workaround for this issue, it solves it for me atleast.
```
    startTimeout() {
        // If autoplay is working we reset timeout and it will never end up inside.
        clearTimeout(this.timer);
        this.timer = setTimeout(() => {
            // This will start play again, important here is to have a timeout that exceeds your "autoplaySpeed". 
            this.slider.innerSlider.play();
        }, 3200);
    }

    render() {
        var settings = {
            infinite: true,
            autoplay: true,
            autoplaySpeed: 3000,
            pauseOnHover: false
        };
        // afterChange and beforeChange docs can be found here: https://github.com/akiran/react-slick
        return (
            <div
                className={ styles.container }
                ref={ item => this.containerRef = item }>
                <Slider
                    ref={ c => this.slider = c }
                    afterChange={ this.startTimeout.bind(this) }
                    beforeChange={ this.startTimeout.bind(this) }
                    {...settings}
                    className={ styles.sliderContainer }>
                    { /*content*/ }
                </Slider>
            </div>
        );
    }
```

**Comment by selenayr** [comment]
please update this issue quickly ,i'm weating for it

**Comment by jarzuaga** [comment]
@akiran is this being worked on?

**Comment by danmo** [comment]
suffering from this issue, any updates ?

**Comment by akiran** [comment]
Fixed this with #651
